WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/PrivacyRing/PrivacyBarViewModel.cs

```diff
@@ -89,69 +86,91 @@ private void CreateSegments(IExtendedList<PrivacyBarItemViewModel> list, IEnumer
 
 	private IEnumerable<PrivacyBarItemViewModel> CreateCoinSegments(IEnumerable<Pocket> pockets, int coinCount)
 	{
-		var total = pockets.Sum(x => Math.Abs(x.Amount.ToDecimal(NBitcoin.MoneyUnit.BTC)));
-		var start = 0.0m;
-
-		var usableWidth = Width - (coinCount * 2);
-
-		var usablePockets =
-				pockets.Where(x => x.Coins.Any())
-					   .OrderByDescending(x => x.Coins.First().HdPubKey.AnonymitySet)
-					   .ToList();
-
-		foreach (var pocket in usablePockets)
+		var totalAmount = pockets.Sum(x => Math.Abs(x.Amount.ToDecimal(MoneyUnit.BTC)));
+		var usableWidth = (decimal)Width - (coinCount - 1) * GapBetweenSegments;
+
+		// Calculate the width of the segments.
+		var rawSegments = pockets
+			.Where(x => x.Coins.Any())
+			.SelectMany(pocket => pocket.Coins)
+			.Select(coin =>
+			{
+				var pocket = pockets.First(pocket => pocket.Coins.Contains(coin));
+				var amount = coin.Amount.ToDecimal(MoneyUnit.BTC);
+				var width = Math.Abs(usableWidth * amount / totalAmount);
+
+				return (OwnerPocket: pocket, Coin: coin, Width: width);
+			}).ToArray();
+
+		// Artificially enlarge segments smaller than the threshold px in order to make them visible.
+		// Meanwhile decrease those segments that are larger than threshold px in order to fit all in the bar.
+		var segmentsToEnlarge = rawSegments.Where(x => x.Width < EnlargeThreshold).ToArray();
+		var segmentsToReduce = rawSegments.Except(segmentsToEnlarge).ToArray();
+		var reduceBy = segmentsToEnlarge.Length * EnlargeBy / segmentsToReduce.Length;
+		if (segmentsToEnlarge.Any() && segmentsToReduce.Length > 0 && segmentsToReduce.All(x => x.Width - reduceBy > 0))
```

Comment:
Nit: Using `.Any()` and `.Length > 0` seems inconsistent. Pick one and use it for both lists

Reply:
Done